### PR TITLE
add tests for views: get (all assets) and post (logic of subscription…

### PR DESCRIPTION
**Pull request by esgaelramos**
… and send mail), create unit tests for models of moneytracker app

### CODE REVIEW

File: tests/test_moneytracker/test_models.py [hunk 1]
```diff
@@ -37,10 +39,60 @@ class AssetModelTest(TestCase):
...
+    Test that a Portfolio object was instantiated successfully using mocks
+    """
+    @patch('moneytracker.models.Portfolio.save')
+    @patch('moneytracker.models.Portfolio.user')
+    def test_portfolio_creation(self, mock_user, mock_save):
+        mock_user.__str__.return_value = "Mocked CustomUser"
+        portfolio_example = Portfolio(user=mock_user)
+
+        self.assertEqual(portfolio_example.user, mock_user)
+        self.assertEqual(str(portfolio_example), f'Portfolio of Mocked CustomUser')
```
Comment (CamilaPua) [inline tests/test_moneytracker/test_models.py:62]: the format

File: tests/test_moneytracker/test_models.py [hunk 2]
```diff
@@ -37,10 +39,60 @@ class AssetModelTest(TestCase):
...
+    @patch('moneytracker.models.PortfolioAsset.asset')
+    def test_portfolioasset_creation(self, mock_asset, mock_portfolio, mock_save):
+        mock_asset.name = "Mocked Asset"
+        mock_portfolio.user.__str__.return_value = "Mocked Portfolio"
+        portfolioasset_example = PortfolioAsset(portfolio=mock_portfolio, asset=mock_asset, quantity=1)
+
+        self.assertEqual(portfolioasset_example.portfolio, mock_portfolio)
+        self.assertEqual(portfolioasset_example.asset, mock_asset)
+        self.assertEqual(portfolioasset_example.quantity, 1)
+        self.assertEqual(str(portfolioasset_example), f'Mocked Asset portfolio of Mocked Portfolio')
```
Comment (CamilaPua) [inline tests/test_moneytracker/test_models.py:81]: the format

File: tests/test_moneytracker/test_models.py [hunk 3]
```diff
@@ -37,10 +39,60 @@ class AssetModelTest(TestCase):
...
+    """
+    @patch('moneytracker.models.Performance.save')
+    @patch('moneytracker.models.Performance.user')
+    def test_performance_creation(self, mock_user, mock_save):
+        mock_user.__str__.return_value = "Mocked CustomUser"
+        performance_example = Performance(user=mock_user, days_to_send_email=1)
+
+        self.assertEqual(performance_example.user, mock_user)
+        self.assertEqual(performance_example.days_to_send_email, 1)
+        self.assertEqual(str(performance_example), f'Performance of Mocked CustomUser')
```
Comment (CamilaPua) [inline tests/test_moneytracker/test_models.py:97]: and the format c: thankkks

**Review by CamilaPua — CHANGES_REQUESTED** [review]

File: tests/test_moneytracker/test_models.py (hunk 3 above)
Comment (esgaelramos) [reply tests/test_moneytracker/test_models.py:97]: im sorry, forget delete the 'f':(

**Review by CamilaPua — APPROVED** [review]

**Comment by CamilaPua** [comment]
0

0
0
0
0

0
0
000  00.  00
 0. yo

000
0000


Obtener Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**